### PR TITLE
filter messages by also looking at media descriptions

### LINKED ISSUE #2284: Filters should be applied to media descriptions
Currently, Tusky doesn’t filter a toot if the word of the filter isn’t in the toot itself or in the CW, but in the media description.

It wasn’t working in Mastodon web when doing client-side filtering until very recently, but that was an oversight and it’s now fixed, see: https://github.com/mastodon/mastodon/pull/17206

* * * *
- Tusky Version: 15.1
- Android Version: 10
- Android Device: Oneplus 5 (cheeseburger)

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by ariasuni**
Fix #2284

I have not tried it yet, I’ll see if I can test that quickly.

### CODE REVIEW

**Comment by Tak** [comment]
Makes sense to me

File: app/src/main/java/com/keylesspalace/tusky/network/FilterModel.kt [hunk 1]
```diff
@@ -29,9 +29,17 @@ class FilterModel @Inject constructor() {
         }
 
         val spoilerText = status.actionableStatus.spoilerText
+        val attachmentsDescriptions = status.attachments
+            .filter { it.description != null }
+            .map { it.description as String }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/network/FilterModel.kt:34]:
```suggestion
            .mapNotNull { it.description }
```

**Review by connyduck — CHANGES_REQUESTED** [review]
Can you please extend the `FilterTest` with a testcase for this?

**Comment by ariasuni** [comment]
I added the test and improved the code as suggested :slightly_smiling_face:

**Comment by connyduck** [comment]
Thank you!

**Pull request merged**